src/hooks/useFeedbackDisclosure.ts

```diff
@@ -0,0 +1,75 @@
+import { useDisclosure } from "@chakra-ui/react"
+import { useLocation } from "react-router-dom"
+
+import { LOCAL_STORAGE_KEYS } from "constants/localStorage"
+
+import { useLoginContext } from "contexts/LoginContext"
+
+import { isEditPageUrl, isSpecialPagesUrl } from "utils/pages"
+
+import { useLocalStorage } from "./useLocalStorage"
+
+type LastSeenFeedbackTime = number
+type UserId = string
+
+type FeedbackStorageMappings = Record<UserId, LastSeenFeedbackTime>
+
+// 1 week in ms
+const NPS_SURVEY_DURATION = 7 * 24 * 60 * 60 * 1000
+
+const { REACT_APP_SHOW_NPS_FORM } = process.env
+
+type UseFeedbackStorageReturn = readonly [number, () => void]
+
+// NOTE: Wrapper to handle get/set of individual keys
+const useFeedbackStorage = (): UseFeedbackStorageReturn => {
+  const { displayedName } = useLoginContext()
+  const [
+    userMappings,
+    setUserMappings,
+  ] = useLocalStorage<FeedbackStorageMappings>(LOCAL_STORAGE_KEYS.Feedback, {})
+
+  const lastSeen = userMappings[displayedName]
+  const setLastSeen = () => {
+    setUserMappings({ ...userMappings, [displayedName]: Date.now() })
+  }
+  // NOTE: `const` cast to infer types properly
+  // so that it knows that the return value is a tuple
+  return [lastSeen, setLastSeen] as const
+}
+
+type UseFeedbackDisclosureReturn = {
+  isOpen: boolean
+  onOpen: () => void
+  onClose: () => void
+}
+export const useFeedbackDisclosure = (): UseFeedbackDisclosureReturn => {
+  const location = useLocation<{ from?: string }>()
+  // NOTE: Despite the typing from the library,
+  // state is NOT guaranteed to exist.
+  const from = location.state?.from ?? ""
+  // NOTE: We show the feedback modal to the users iff
+  // they are navigating away from the editor
+  const isLeavingContentPage = isEditPageUrl(from) || isSpecialPagesUrl(from)
+  const shouldShowNpsForm = REACT_APP_SHOW_NPS_FORM?.trim() === "true"
+
+  const [lastSeen, setLastSeen] = useFeedbackStorage()
+  // NOTE: Either this is the first time the user has ever seen the survey
+  // or that the user has seen the survey but it has been more than 3 weeks.
+  // Because we toggle the survey on every month, this indicates that they should
```

Comment:
hmm shall we be precise on the frequency haha 
7 days vs 3 weeks vs 1 mth

Reply:
2 things here - frequency of survey is 1 month, duration is 1 wek